fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -1,108 +1,140 @@
 """
 Finds leaks between splits.
+
+| Copyright 2017-2024, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
 """
+import logging
 
-from copy import copy
-import warnings
+import eta.core.utils as etau
 
-from fiftyone import ViewField as F
 import fiftyone.core.brain as fob
-import fiftyone.brain.similarity as sim
-import fiftyone.core.utils as fou
 import fiftyone.core.validation as fov
+import fiftyone.zoo as foz
+from fiftyone import ViewField as F
+
+import fiftyone.brain as fb
+import fiftyone.brain.similarity as fbs
+import fiftyone.brain.internal.core.utils as fbu
+
 
-fbu = fou.lazy_import("fiftyone.brain.internal.core.utils")
+logger = logging.getLogger(__name__)
 
 _DEFAULT_MODEL = "resnet18-imagenet-torch"
+_DEFAULT_BATCH_SIZE = None
 
 
 def compute_leaky_splits(
     samples,
-    brain_key=None,
-    split_views=None,
-    split_field=None,
     split_tags=None,
-    threshold=0.2,
-    similarity_brain_key=None,
-    embeddings_field=None,
+    split_field=None,
+    split_views=None,
+    threshold=None,
+    embeddings=None,
+    similarity_index=None,
     model=None,
     model_kwargs=None,
-    similarity_backend=None,
-    similarity_config_dict=None,
-    **kwargs,
+    batch_size=None,
+    num_workers=None,
+    skip_failures=True,
+    progress=None,
 ):
     """See ``fiftyone/brain/__init__.py``."""
 
-    fov.validate_collection(samples)
+    fov.validate_image_collection(samples)
+
+    if etau.is_str(embeddings):
```

Comment:
> what do you think about having an EmbeddingsBrainMethodMixin or something similar that handles the boilerplate around embeddings and models? It seems strange to me to duplicate that code every time.

Taking this conclusion to the logical extreme, the way to implement leaky splits with the absolute minimum amount of code would definitely have been to add it to the `DuplicatesMixin` interface, say like this:

```py
index = fob.compute_similarity(dataset, ...)

index.find_leaks(splits=["train", "test"], threshold=0.2)
index.find_leaks(splits="split_field", threshold=0.2)
index.find_leaks(splits={"train": train_view, "test": test_view", threshold=0.2)

print(index.leak_ids)  # ['XXXX', ...]

leaks = index.leaks_view()
no_leaks_view = index.no_leaks_view()
leaks_for_sample = index.leaks_for_sample(index.leak_ids[0])
```

After all, leaky splits is really just an add-on on top of the duplicates stuff, and that's exactly how those were implemented:

```py
index.find_duplicates()
index.duplicate_ids
index.duplicates_view()

index.find_unique()
index.unique_ids
index.unique_view()
```

Now having said that, I do see value in having a top-level `compute_leaky_splits()` method, as that's more of a use-case-centric naming convention (methods based on what you want to do), rather than an implementation-centric naming convention (methods based on how they work).

So, I'm onboard with `compute_leaky_splits()` and in fact think it could make sense in a follow-up PR to add a `compute_near_duplicates()` method that basically wraps `DuplicatesMixin.find_duplicates()`.

So, to your question: the only reason for the boilerplate code here is to parse out the JSON serializable config parameters to pass to `BrainMethod Config`. **BUT**, as discussed elsewhere on this PR, there's really not any need to do all that as currently implemented because there's no `brain_key`.

I'll make a follow-up PR that prunes the unnecessary boilerplate to see what we think about it.

Reply:
Follow-up PR is here: https://github.com/voxel51/fiftyone-brain/pull/213